### PR TITLE
Support providing option channel types as list

### LINKED ISSUE #991: multiple-type arg option is beak
### Summary

Cannot set multiple type for arg in slash command

### Reproduction Steps

```python
@slash_command()
async def test(self, ctx, channel: Option([discord.TextChannel, discord.VoiceChannel])):
    print(channel)
```

### Minimal Reproducible Code

```python
Traceback (most recent call last):
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python310\lib\site-packages\discord\cog.py", line 705, in _load_from_module_spec
    spec.loader.exec_module(lib)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "c:\Users\Andrew\Documents\tuflya\cogs\sendmsg.py", line 8, in <module>
    class sendmsg(commands.Cog):
  File "c:\Users\Andrew\Documents\tuflya\cogs\sendmsg.py", line 14, in sendmsg
    channel: Option([discord.TextChannel, discord.VoiceChannel], "\u0412\u044b\u0431\u0435\u0440\u0438\u0442\u0435 \u043a\u0430\u043d\u0430\u043b"),
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python310\lib\site-packages\discord\commands\options.py", line 72, in __init__
    _type = SlashCommandOptionType.from_datatype(input_type)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python310\lib\site-packages\discord\enums.py", line 670, in from_datatype
    if datatype.__name__ in ["Member", "User"]:
AttributeError: 'list' object has no attribute '__name__'. Did you mean: '__ne__'?
```

### Expected Results

channel arg should accept both types

### Actual Results

can't use multiple-type in slash command

### Intents

default, members

### System Information

Python Version: 3.10
Pycord Version: 2.0.0b4

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Ratery**
## Summary
Support providing option channel types as a list like `[discord.TextChannel, discord.VoiceChannel]`.
Fixed #991.

## Checklist
- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Ratery** [comment]
> I think we should force one type just like most of the other code.

Ok, but then we should update the [example](https://github.com/Pycord-Development/pycord/blob/429dadce242bcdd8cfc224a072a5558497d0618b/examples/app_commands/slash_options.py#L25)

**Comment by Ratery** [comment]
So, we should update examples or change code of the library because it accepts only tuples but lists were used in that example.

**Comment by Dorukyum** [comment]
What I've previously meant is that maybe we could check for `hasattr(passed, '__iter__')` or some other way of checking for an iterable. Open for discussion.

**Comment by Dorukyum** [comment]
Let's just use `isinstance(datatype, list)` for now, it would be more consistent with the rest of the library.

File: discord/enums.py [hunk 1]
```diff
@@ -658,7 +658,7 @@ class SlashCommandOptionType(Enum):
 
     @classmethod
     def from_datatype(cls, datatype):
-        if isinstance(datatype, tuple):  # typing.Union has been used
+        if isinstance(datatype, (tuple, list)):  # typing.Union has been used
```
Comment (Dorukyum) [inline discord/enums.py:661]: I don't think this should be changed, the comment states that Union is converted to a tuple.
```suggestion
        if isinstance(datatype, tuple):  # typing.Union has been used
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Dorukyum — APPROVED** [review]

File: discord/enums.py (hunk 1 above)
Comment (krittick) [reply discord/enums.py]: This still causes the error from #991 unless you change it to include both.

**Comment by Dorukyum** [comment]
> We should use typing.Union[]

Union support has been added to #1251.

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**